### ISSUE TITLE
Speaker Verification: All Speakers Getting Perfect 1.000 Similarity Scores

### ISSUE THREAD

**Issue by misterpathologist** [huggingface/transformers#36124]
### Bug Report

Model name (e.g. bert-base-cased): pyannote/embedding
Language (if applicable): English
Framework (PyTorch, TensorFlow, etc...): PyTorch

### Description

Using pyannote/embedding for speaker verification, getting perfect similarity scores (1.000) for all speakers, even between obviously different voices in an audiobook.

### Code To Reproduce The Issue
python
import torch
import torchaudio
from pyannote.audio import Model
import torch.nn.functional as F
Setup
device = torch.device("cuda")
embedding_model = Model.from_pretrained("pyannote/embedding",
use_auth_token='xxx').to(device)
Load and process reference audio
reference_waveform, sample_rate = torchaudio.load("reference.flac")
reference_waveform = reference_waveform.mean(dim=0, keepdim=True).to(device)
reference_features = embedding_model(reference_waveform.unsqueeze(0))
reference_features = F.normalize(reference_features, p=2, dim=1)
Load test audio segment
test_waveform, = torchaudio.load("test.flac")
test_waveform = test_waveform.mean(dim=0, keepdim=True).to(device)
test_embedding = embedding_model(test_waveform.unsqueeze(0))
test_embedding = F.normalize(test_embedding, p=2, dim=1)
Calculate similarity
similarity = F.cosine_similarity(reference_features, test_embedding, dim=1).mean()
print(f"Similarity: {similarity.item():.6f}")

### Expected Results

Different speakers should have varying similarity scores below 1.000

### Actual Results

All speakers get perfect 1.000 similarity scores:
- Speaker A vs Reference: 1.000000
- Speaker B vs Reference: 0.999998
- Speaker C vs Reference: 1.000000

### Environment
- pyannote.audio: 3.1.1
- torch: 2.5.1+cu124
- Platform: Google Colab (Ubuntu Linux)
- CUDA: Yes
- GPU: Tesla T4
- Python: 3.11
- torchaudio: 2.5.1+cu124

### Additional Context

- Using professional audiobook with distinct voices
- Reference is 10-minute high-quality audio
- Testing with 4-hour audiobook
- Consistent 1.000 similarity across all different speakers

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

1. Install dependencies:
   pip install pyannote.audio==3.1.1 torch==2.5.1+cu124 torchaudio==2.5.1+cu124

2. Use reference audio (10-minute FLAC file) and test audio (different speaker, FLAC file)

3. Run the provided code:
   - Load model and audio files
   - Extract embeddings
   - Calculate similarity
   
4. Observe that similarity scores are always 1.000 regardless of speaker differences

Full code provided in the description above. This can be reproduced with any two different speakers' audio files.

### Expected behavior

The similarity scores should:
- Be less than 1.000 for different speakers
- Show variation between different voices
- Have lower scores for more dissimilar voices
- Only approach 1.000 for the same speaker

Instead, we're getting perfect 1.000 similarity scores for all speakers, even between obviously different voices (male/female) from a professional audiobook.

**Comment by eustlb** [comment]
Thanks for opening this issue!
Not able to reproduce here without the specific audio. Therefore closing this issue cleanly. @misterpathologist feel free to ping here with a self-contained reproducer, happy to take a look then 🤗
